adapters/repos/db/migrator.go

```diff
@@ -208,3 +211,34 @@ func (m *Migrator) UpdateInvertedIndexConfig(ctx context.Context, className stri
 
 	return idx.updateInvertedIndexConfig(ctx, conf)
 }
+
+func (m *Migrator) RecalculateVectorDimensions(ctx context.Context) error {
+	count := 0
+	m.logger.
+		WithField("action", "reindex").
+		Info("Reindexing dimensions, this may take a while")
+
+	// Iterate over all indexes
+	for _, index := range m.db.indices {
+		// Iterate over all shards
+
+		if err := index.IterateObjects(ctx, func(index *Index, shard *Shard, object *storobj.Object) error {
+			count = count + 1
+			err := shard.extendDimensionTrackerLSM(len(object.Vector), object.DocID())
+			return err
+		}); err != nil {
+			return err
+		}
+	}
+	go func() {
+		for {
+
+			m.logger.
+				WithField("action", "reindex").
+				Warn(fmt.Sprintf("Reindexed %v objects.  Reindexing dimensions complete.  Please remove environment variable REINDEX_VECTOR_DIMENSIONS_AT_STARTUP before next startup", count))
```

Comment:
I think that you can change double spaces: `  ` to just one space: ` ` and use a built-in `Warnf` method instead of `Warn` + `fmt.Sprintf`